### PR TITLE
CI: mdbook added to pipeline + Build badge

### PR DESCRIPTION
**Pull request by 4ydan**
What was done in this PR:
- Added mdbook to the pipeline
- Created a buildbage in the root README.md
- Small refactoring in the pipeline script(renaming all 'check' to 'test')
- Two extra lines in .prettierignore which are not relevant to this PR but might be handy for later on.

Issue #275.

## Basics

- [x] I added a line to [/doc/CHANGELOG.md](/doc/CHANGELOG.md)
- [x] The PR is rebased with current master.
- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.

## Checklist

- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by kitzbergerg** [comment]
From what I've seen it looks good. For the details about how different stages should be build up I'd refer to @lukashartl (e.g. if mdbook test and build should be in the same stage).
Apart from that the rustdoc as described in #275 is missing and the deployment of both docs as well. You mentioned that this is still in progress, so I would suggest marking this PR as a draft until you think it can be merged.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,45 @@
+export VITE_BASE_API_URL=""
+export VITE_NEXTCLOUD_URI="https://cloud.permaplant.net"
+
+
```
Comment (markus2330) [inline Makefile:4]:
```suggestion
.PHONY: build-frontend
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,45 @@
+export VITE_BASE_API_URL=""
+export VITE_NEXTCLOUD_URI="https://cloud.permaplant.net"
+
```
Comment (markus2330) [inline Makefile:3]: Can you also provide an "all" or "run-all" target?

File: ci/build-scripts/build-storybook.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+echo "checking storybook"
```
Comment (markus2330) [inline ci/build-scripts/build-storybook.sh:3]:
```suggestion
echo "building storybook"
```

**Review by markus2330 — APPROVED** [review]
Great to have these many improvements in CI and for local builds/tests! Looking forward to see this merged soon.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+build-frontend: test-frontend
+	cd frontend && npm run build
+
+.PHONY: test
+test: test-frontend
+
+.PHONY: test-frontend
+test-frontend:
+	cd frontend && npm ci
```
[the remark this answers is not in the document]
Comment (4ydan) [reply Makefile:13]: The idea was somehow to maybe use the makefile in the jenkins pipeline

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,45 @@
+export VITE_BASE_API_URL=""
+export VITE_NEXTCLOUD_URI="https://cloud.permaplant.net"
+
+
+build-frontend: test-frontend
+	cd frontend && npm run build
```
[the remark this answers is not in the document]
Comment (4ydan) [reply Makefile:6]: yes, thank you

**Comment by 4ydan** [comment]
Ok, so I removed all the makefile stuff now to keep this PR small and make it fullfil what was actually requested in the issue. I will open another PR making the makefile suggestions and more.

**Comment by markus2330** [comment]
Yes, lets add the Makefile in another PR. Looking forward to it! :heart:

**Pull request merged**